internal/host/transition_test.go

```diff
@@ -5988,3 +5989,176 @@ func validateEqualProgress(p1, p2 *models.HostProgressInfo) {
 		Expect(time.Time(p1.StageUpdatedAt).Equal(time.Time(p2.StageUpdatedAt))).To(BeTrue())
 	}
 }
+
+type testState struct {
+	state string
+}
+
+func newTestState(state string) *testState {
+	return &testState{
+		state: state,
+	}
+}
+
+func (ts *testState) State() stateswitch.State {
+	return stateswitch.State(ts.state)
+}
+
+func (ts *testState) SetState(state stateswitch.State) error {
+	ts.state = string(state)
+	return nil
+}
+
+var allValidationIDs []validationID = []validationID{
+	IsMediaConnected,
+	IsConnected,
+	HasInventory,
+	IsMachineCidrDefined,
+	BelongsToMachineCidr,
+	HasMinCPUCores,
+	HasMinValidDisks,
+	HasMinMemory,
+	HasCPUCoresForRole,
+	HasMemoryForRole,
+	IsHostnameUnique,
+	IsHostnameValid,
+	IsIgnitionDownloadable,
+	BelongsToMajorityGroup,
+	IsPlatformNetworkSettingsValid,
+	IsNTPSynced,
+	SucessfullOrUnknownContainerImagesAvailability,
+	AreLsoRequirementsSatisfied,
+	AreOdfRequirementsSatisfied,
+	AreCnvRequirementsSatisfied,
+	AreLvmRequirementsSatisfied,
+	SufficientOrUnknownInstallationDiskSpeed,
+	HasSufficientNetworkLatencyRequirementForRole,
+	HasSufficientPacketLossRequirementForRole,
+	HasDefaultRoute,
+	IsAPIDomainNameResolvedCorrectly,
+	IsAPIInternalDomainNameResolvedCorrectly,
+	IsAppsDomainNameResolvedCorrectly,
+	CompatibleWithClusterPlatform,
+	IsDNSWildcardNotConfigured,
+	DiskEncryptionRequirementsSatisfied,
+	NonOverlappingSubnets,
+	VSphereHostUUIDEnabled,
+	CompatibleAgent,
+	NoSkipInstallationDisk,
+	NoSkipMissingDisk,
+	HostValidationIDServiceHasSufficientSpokeKubeAPIAccess,
+}
+
+var allConditions []conditionId = []conditionId{
+	InstallationDiskSpeedCheckSuccessful,
+	ClusterPreparingForInstallation,
+	ClusterPendingUserAction,
+	ClusterInstalling,
+	ValidRoleForInstallation,
+	StageInWrongBootStages,
+	ClusterInError,
+	SuccessfulContainerImageAvailability,
+}
```

Comment:
I have somehow hard time believing that the next person adding new validation and/or condition will remember to add it here. Can we invent any mechanism to somehow ensure/enforce that? If no that's okay, but I just want to raise awareness here

Reply:
No clean way of doing that that I can think of